app/api/toc_generation/tocService.ts

```diff
@@ -0,0 +1,46 @@
+import { files, uploadsPath } from 'api/files';
+import { prettifyError } from 'api/utils/handleError';
+import errorLog from 'api/log/errorLog';
+import request from 'shared/JSONRequest';
+import entities from 'api/entities';
+
+const tocService = (serviceUrl: string) => ({
+  async processNext() {
+    const [nextFile] = await files.get(
+      {
+        $or: [{ toc: { $size: 0 } }, { toc: { $exists: false } }],
+        type: 'document',
+        filename: { $exists: true },
+      },
+      '',
+      {
+        sort: { _id: 1 },
+        limit: 1,
+      }
+    );
+
+    try {
+      if (nextFile) {
+        const toc = await request.uploadFile(
+          serviceUrl,
+          nextFile.filename,
+          uploadsPath(nextFile.filename)
+        );
+        await files.save({ ...nextFile, toc, generatedToc: true });
```

Comment:
Should this line fail, the error log will log the error, but subsequent calls will fetch the same file over and over again.  Do we want that to be the case? If this is only logging errors and not failing, how is the caller supposed to know something went wrong?

Reply:
it is not supposed to, error logs should be enough for now, we designed a different system to handle this kinds of problems, but it was too complex for this initial testing version. the main reason the error is handled here is to prevent a crash on "service not online" error as this runs outside the normal request/response flow.

do you have any suggestion to quickly make this more resilient ?

Reply:
@RafaPolit "better" error handling is now implemented, this is basic error handling taking into account this is a beta feature.

1. when service returns empty toc: a fake toc is generated labeled "empty toc was generated"
2. when any error except "service not available": a fake toc is generated labeled with the error message
3. when service not available: does nothing, keeps trying until it is available.